client/state/current-user/actions.js

```diff
@@ -15,3 +18,15 @@ export function setCurrentUser( user ) {
 		user,
 	};
 }
+
+export async function redirectToLogout( postLogoutRedirectUrl ) {
```

Comment:
This action could be in `state/logout/actions`. That module already contains a `logoutUser` action creator.

Which does the logout in a slightly different way: posting a REST XHR request to the `wp-login.php?action=logout-endpoint` instead of top-level redirecting to `wp-login.php?action=logout`.

I'm not sure what exactly are the pros and cons of each approach, and we should unify them eventually. The REST XHR approach is going to be troublesome with 3rd party cookies blocked though. It won't really log you out (i.e., remove the auth cookies) on a non-`wordpress.com` domain.

Reply:
> This function can't be async, only the returned thunk can. 

I'm not sure how this happened, but thanks for catching it. Fixed in https://github.com/Automattic/wp-calypso/pull/53661/commits/3774400ac1c2252cdc7571e2fbcd0cb37ac523d2

Reply:
> This action could be in `state/logout/actions`. That module already contains a `logoutUser` action creator.
> 
> Which does the logout in a slightly different way: posting a REST XHR request to the `wp-login.php?action=logout-endpoint` instead of top-level redirecting to `wp-login.php?action=logout`.
> 
> I'm not sure what exactly are the pros and cons of each approach, and we should unify them eventually. The REST XHR approach is going to be troublesome with 3rd party cookies blocked though. It won't really log you out (i.e., remove the auth cookies) on a non-`wordpress.com` domain.

Let's leave that unification for another time. These have been separate before and this PR's intention isn't to change that.